src/Stan.hs

```diff
@@ -25,11 +25,11 @@ stanPlugin :: [CommandLineOption] -> TcGblEnv -> HsGroup GhcRn -> TcM (TcGblEnv,
 stanPlugin _ tcEnv hsGroup = do
     -- let modulePath = ms_hspp_file modSummary
     let warnings = stan hsGroup
-    unless (null warnings) $ do
+    unless (null warnings) $ liftIO $ do
         -- print the module name
         let fileName = unpackFS $ srcSpanFile $ tcg_top_loc tcEnv
-        liftIO $ putStrLn $ "!!! Found " ++ show (length warnings) ++ " warning(s) in " ++ fileName
-        fileContent <- liftIO $ readFile fileName
-        for_ warnings (liftIO . putStrLn . showWarning fileContent)
+        putStrLn $ moduleOverall warnings fileName
```

Comment:
It was a good idea to move this into separate function. Sometimes functions are needed not only to reduce code duplication but to improve code readability and to assign name (description) to some chunk of code 👍